platform/detect_inputs.go

```diff
@@ -7,59 +7,5 @@ import (
 // DefaultDetectInputs accepts a Platform API version and returns a set of lifecycle inputs
 // with default values filled in for the `detect` phase.
 func DefaultDetectInputs(platformAPI *api.Version) LifecycleInputs {
-	var inputs LifecycleInputs
-	switch {
-	case platformAPI.AtLeast("0.11"):
-		inputs = defaultDetectInputs()
-	case platformAPI.AtLeast("0.10"):
-		inputs = defaultDetectInputs010()
-	case platformAPI.AtLeast("0.6"):
-		inputs = defaultDetectInputs06To09()
-	case platformAPI.AtLeast("0.5"):
-		inputs = defaultDetectInputs05()
-	default:
-		inputs = defaultDetectInputs03To04()
-	}
-	inputs.PlatformAPI = platformAPI
-	return inputs
-}
-
-func defaultDetectInputs() LifecycleInputs {
```

Comment:
Just updated detect for now, but we can remove the individual constructors for each phase if we like this pattern